components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataDeleteRecordsTask.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.application.authentication.framework.store;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.concurrent.BlockingDeque;
+
+public class SessionDataDeleteRecordsTask implements Runnable {
+
+    private static final Log log = LogFactory.getLog(SessionDataDeleteRecordsTask.class);
+    private BlockingDeque<SessionContextDO> sessionContextDeleteTempQueue;
+
+    public SessionDataDeleteRecordsTask(BlockingDeque<SessionContextDO> sessionContextDeleteTempQueue) {
+        this.sessionContextDeleteTempQueue = sessionContextDeleteTempQueue;
+    }
+
+    @Override
+    public void run() {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Session Context delete temp records consumer is started");
+        }
+
+        while (true) {
```

Comment:
Shall we keep a status variable which is assigned false while the component is deactivated, so that we will not see any unexpected stack traces while server is shutting down ?